docs/testplan/Smartswitch-test-plan.md

```diff
@@ -645,6 +672,673 @@ root@sonic:/home/cisco# show chassis modules status
+ * Append to the existing test case: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/platform_tests/test_power_off_reboot.py
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+
+After power cycle of the testbed, 
+
+```
+root@sonic:/home/cisco#
+root@sonic:/home/cisco#
+.
+(Going for power off reboot using pdu controller)
+.
+root@sonic:/home/cisco#
+root@sonic:/home/cisco#
+root@sonic:/home/cisco# show chassis modules status
+  Name           Description    Physical-Slot    Oper-Status    Admin-Status           Serial
+------  --------------------  ---------------  -------------  --------------  ---------------
+  DPU0  Data Processing Unit              N/A        Offline            down  154226463179136
+  DPU1  Data Processing Unit              N/A        Offline            down  154226463179152
+  DPU2  Data Processing Unit              N/A        Offline            down  154226463179168
+  DPUX  Data Processing Unit              N/A        Offline            down  154226463179184
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# 
+root@sonic:/home/cisco# config chassis startup DPU0
+root@sonic:/home/cisco# config chassis startup DPU1
+root@sonic:/home/cisco# config chassis startup DPU2
+root@sonic:/home/cisco# config chassis startup DPUX
+root@sonic:/home/cisco#
+root@sonic:/home/cisco#
+root@sonic:/home/cisco# config chassis modules startup <DPU_NUMBER>
+root@sonic:/home/cisco#
+root@sonic:/home/cisco# show chassis modules status
+  Name           Description    Physical-Slot    Oper-Status    Admin-Status           Serial
+------  --------------------  ---------------  -------------  --------------  ---------------
+  DPU0  Data Processing Unit              N/A         Online              up  154226463179136
+  DPU1  Data Processing Unit              N/A         Online              up  154226463179152
+  DPU2  Data Processing Unit              N/A         Online              up  154226463179168
+  DPUX  Data Processing Unit              N/A         Online              up  154226463179184
+
+root@sonic:/home/cisco# ping 169.254.200.1
+PING 169.254.200.1 (169.254.200.1) 56(84) bytes of data.
+64 bytes from 169.254.200.1: icmp_seq=1 ttl=64 time=0.160 ms
+^C
+--- 169.254.28.1 ping statistics ---
+1 packets transmitted, 1 received, 0% packet loss, time 0ms
+rtt min/avg/max/mdev = 0.160/0.160/0.160/0.000 ms
+root@sonic:/home/cisco# 
+```
+
+#### Pass/Fail Criteria
+ * Verify number of DPUs from inventory file for the testbed and number of DPUs shown in the cli output.
+ * Verify Ping works to all the mid plane ip listed in the ansible inventory file for the testbed.
+
+
+### 1.21 Check DPU status and pcie Link after SW trip by temperature trigger
+
+#### Steps
+
+ * Infrastructure will be provided to run the scripts that triggers the temperature trip based on vendor.
+ * The following is the example sequence to trigger temperature trip on the dpu
```

Comment:
Do we have any API which can call vendor specific implementation?

Reply:
It will be implemented as part of test scripts. The plan is to have function that takes vendor specific scripts to be run and checks the output. 

There is not existing api implementation.